lib/active_model/serializer/configuration.rb

```diff
@@ -21,13 +21,16 @@ def config.array_serializer
 
         config.default_includes = '*'
         config.adapter = :attributes
+        config.key_transform = nil
         config.jsonapi_resource_type = :plural
+        config.jsonapi_namespace_separator = '--'.freeze
+        # http://jsonapi.org/recommendations/#naming
+        config.jsonapi_type_transform = :dashed
```

Comment:
the type transform was (I thought) was there in the other PR. I like the differentiation between type and key, but this is what introduces the breaking change.

key_transform by default is nil (unless your json api, then it's dash)
type_transform is dash.

in the old implementation, it was using both key_transform and type_transform, which is why tests were failing, I think. key_transform didn't matter because type_transform was overriding whatever key_transform wanted to do